### PR TITLE
added several plan building speed improvements

### PR DESCRIPTION
**Pull request by daniel-shields**

### CODE REVIEW

File: noxfile.py [hunk 1]
```diff
@@ -30,6 +30,7 @@ def install_with_constraints(session, *args, **kwargs):
             "poetry",
             "export",
             "--dev",
+            "--without-hashes",
```
Comment (daniel-shields) [inline noxfile.py:33]: I think this is the correct action based on https://github.com/python-poetry/poetry/issues/3472 (search "without-hashes")

File: poetry.lock [hunk 2]
```diff
@@ -1,258 +1,250 @@
 [[package]]
-category = "dev"
-description = "A configurable sidebar-enabled Sphinx theme"
```
Comment (daniel-shields) [inline poetry.lock:3]: This seemingly just bumps a bunch of versions...

File: src/uberjob/_execution/run_physical.py [hunk 3]
```diff
@@ -88,24 +92,24 @@ def prep_run_physical(
         plan, output_node
     )
     plan = prune_source_literals(plan, inplace=inplace)
+    on_started = on_started or identity
```
Comment (daniel-shields) [inline src/uberjob/_execution/run_physical.py:95]: This avoids three null checks for each node...

File: src/uberjob/_execution/run_physical.py [hunk 4]
```diff
@@ -88,24 +92,24 @@ def prep_run_physical(
...
-                bound_call = bound_call_lookup[node]
                 retry(bound_call.value.run)()
-                bound_call.value = None
             except Exception as e:
-                if on_failed is not None:
-                    on_failed(node, e)
+                on_failed(node, e)
                 raise
+            finally:
+                bound_call.value = None
```
Comment (daniel-shields) [inline src/uberjob/_execution/run_physical.py:110]: Moving this to the finally block fixes a memory leak related to failures. In practice this probably didn't matter.

File: src/uberjob/_execution/run_physical.py [hunk 5]
```diff
@@ -55,7 +55,7 @@ def _create_bound_call_lookup_and_output_slot(
     plan: Plan, output_node: Optional[Node] = None
 ):
     result_lookup = {
-        node: Slot(node.value if type(node) is Literal else None)
+        node: node if type(node) is Literal else Slot(None)
```
Comment (daniel-shields) [inline src/uberjob/_execution/run_physical.py:58]: This is a tiny step towards eliminating Slot from run_physical entirely, which is possible but I don't want to do as part of this PR.

File: pyproject.toml [hunk 6]
```diff
@@ -32,6 +32,7 @@ isort = "^5.5.3"
 pytest = "^5.2"
 pytest-cov = "^2.10.1"
 releases = "^1.6.3"
+six = "^1.15.0"
```
Comment (daniel-shields) [inline pyproject.toml:35]: This addresses an issue with CI.

**Review by timothy-shields — APPROVED** [review]

**Pull request merged**